### PR TITLE
Set type of location list items for diagnostics

### PR DESCRIPTION
**Pull request by rhysd**
Item of location list can have key `"type"`, which represents single-character error type, 'E', 'W', etc. This can be calculated from severity of diagnostics results but currently it is not set.

This PR sets the key as follows:

- `"E"` for error
- `"W"` for warning
- `"I"` for information
- `"H"` for hint
- does not set `"type"` when no severity is specified in the diagnostics results

With this PR, location list now shows the type of item:

- before:

スクリーンショット 2021-04-12 17 45 49

- after:

スクリーンショット 2021-04-12 17 46 42

### CODE REVIEW

File: autoload/lsp/ui/vim/utils.vim [hunk 1]
```diff
@@ -110,24 +110,30 @@ function! lsp#ui#vim#utils#diagnostics_to_loc_list(result) abort
...
+            let l:location_item = {
                 \ 'filename': l:path,
                 \ 'lnum': l:line,
                 \ 'col': l:col,
                 \ 'text': l:text,
-                \ })
+                \ }
+            if l:has_severity
+                " 'E' for error, 'W' for warning, 'I' for information, 'H' for hint
+                let l:location_item['type'] = s:diagnostic_severity[l:item['severity']][0]
```
Comment (mattn) [inline autoload/lsp/ui/vim/utils.vim:134]: Should be `s:get_diagnostic_severity_text(l:item['severity'])[0]` ?

Comment (rhysd) [reply autoload/lsp/ui/vim/utils.vim:134]: I apologize for the delay. I was focusing on vim-lsp-ale.

> Should be s:get_diagnostic_severity_text(l:item['severity'])[0] ?

That sounds better. And I noticed that I can remove duplicate in the code by storing the severity text to a variable. I'll update this branch.

**Review by mattn — APPROVED** [review]

**Comment by mattn** [comment]
Feel free to merge. Thanks.

**Comment by rhysd** [comment]
Thank you for merging the upstream. I'm merging.

**Pull request merged**